### PR TITLE
DBZ-3265 using plugin to obtain git revision

### PR DESCRIPTION
**Pull request by mdrillin**
utilizing plugin to obtain the git commit hash for the build

### CODE REVIEW

**Review by indraraj — APPROVED** [review]
Tested local and container builds, looks fine to me.

File: ui/packages/ui/webpack.common.js [hunk 1]
```diff
@@ -3,10 +3,12 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
 const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
 const webpack = require('webpack');
 const { dependencies } = require("./package.json");
+const GitRevisionPlugin = require('git-revision-webpack-plugin');
+const gitRevisionPlugin = new GitRevisionPlugin({commithashCommand: 'rev-parse --short HEAD'});
```
Comment (rk3rn3r) [inline ui/packages/ui/webpack.common.js:7]:
```suggestion
const gitRevisionPlugin = new GitRevisionPlugin({versionCommand: 'describe --always --tags'});
```

File: ui/packages/ui/webpack.common.js [hunk 2]
```diff
@@ -3,10 +3,12 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
...
 
 // Try the environment variable, otherwise use root
 const ASSET_PATH = process.env.ASSET_PATH || '/';
-const COMMIT_HASH = process.env.COMMIT_HASH || require('child_process').execSync('git rev-parse --short HEAD').toString().trim();
+const COMMIT_HASH = process.env.COMMIT_HASH || gitRevisionPlugin.commithash();
```
Comment (rk3rn3r) [inline ui/packages/ui/webpack.common.js:11]:
```suggestion
const COMMIT_HASH = process.env.COMMIT_HASH || gitRevisionPlugin.version();
```

Comment (rk3rn3r) [reply ui/packages/ui/webpack.common.js:11]: We can capture the latest tag (that build is based on, plus the short hash) like this:
```
$ git describe --always --tags
v1.5.0.Beta2-2-g3cf30f2
```

File: ui/packages/ui/webpack.common.js [hunk 3]
```diff
@@ -20,8 +22,6 @@ module.exports = {
     // This makes it possible for us to safely use env vars on our code
     new webpack.DefinePlugin({
       'process.env.ASSET_PATH': JSON.stringify(ASSET_PATH),
-    }),
-    new webpack.DefinePlugin({
       'process.env.COMMIT_HASH': JSON.stringify(COMMIT_HASH),
```
Comment (rk3rn3r) [inline ui/packages/ui/webpack.common.js:25]:
```suggestion
      'process.env.COMMIT_VERSION': JSON.stringify(COMMIT_VERSION),
```

File: ui/packages/ui/webpack.common.js (hunk 2 above)
Comment (rk3rn3r) [reply ui/packages/ui/webpack.common.js:11]: What happens here, when git is not available? Did you test that (removing `git` from the `microdnf install git` call in the builder part of the Dockerfile? Do we have a proper error message?

Comment (uidoyen) [reply ui/packages/ui/webpack.common.js:11]: It looks like we need `git` installed (see https://github.com/pirelenito/git-revision-webpack-plugin/blob/v3/lib/helpers/run-git-command.js#L9)

Comment (rk3rn3r) [reply ui/packages/ui/webpack.common.js:11]: I know. But the question is what happens when it's not available. I expect the build to fail, and not, like before, continue without assets.

Comment (uidoyen) [reply ui/packages/ui/webpack.common.js:11]: I tried after removing `git` from  `Dockerfile` and build was success this time.

Comment (rk3rn3r) [reply ui/packages/ui/webpack.common.js:11]: And what is the COMMIT_HASH then or is it just empty? This should not happen. When the hash (and git) is unavailable the build should fail. WDYT?

Comment (uidoyen) [reply ui/packages/ui/webpack.common.js:11]: Yes, the build should fail when `git` is unavailable but build is success and I am getting`COMMIT_HASH` is `3762ffd`.

Comment (rk3rn3r) [reply ui/packages/ui/webpack.common.js:11]: @uidoyen So what is your local `git rev-parse --short HEAD`? Is it the correct value? And why does it "work"?

**Comment by indraraj** [comment]
@mdrillin I think this can be closed since @uidoyen  is working to use the GitHub API to fetch the commit id which don't need git to be locally installed.

**Pull request closed without merging**